### PR TITLE
[docs] Add documentation for KV API (how tos and explainer).

### PR DESCRIPTION
**Pull request by iamveritas**
[docs] Add documentation for KV API (how tos and explainer).
I also re-arranged the items listed under main `How To` section for Key-Value to be listed before multi-index (it is decided we want to promote the KV as the primary API).

### CODE REVIEW

File: docs/06_how-to-guides/30_key-value-api/40_how-to-delete-from-table.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        &person::account_name };
+
+     address_table(name contract_name) {
+        init(contract_name, account_name_uidx)
+     }
+  };
+  public:
+     using contract::contract;
+
+     // creates if not exists, or updates if already exists, a person
```
Comment (kimjh2005) [inline docs/06_how-to-guides/30_key-value-api/40_how-to-delete-from-table.md:76]: comment from upsert?

File: docs/06_how-to-guides/30_key-value-api/40_how-to-delete-from-table.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+     [[eosio::action]]
+     void delete(eosio::name account_name,
```
Comment (kimjh2005) [inline docs/06_how-to-guides/30_key-value-api/40_how-to-delete-from-table.md:78]: Only index field needed to delete.

File: docs/06_how-to-guides/30_key-value-api/30_how-to-upsert-into-table.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+// creates if not exists, or updates if already exists, a person
+[[eosio::action]]
+void kv_addr_book::upsert(
+     eosio::name account_name,
+     string first_name,
+     string last_name,
+     string personal_id) {
+  address_table addresses{"kvaddrbook"_n};
+
+  // upsert into kv table
```
Comment (kimjh2005) [inline docs/06_how-to-guides/30_key-value-api/30_how-to-upsert-into-table.md:100]: upsert?

Comment (iamveritas) [reply docs/06_how-to-guides/30_key-value-api/30_how-to-upsert-into-table.md:100]: yeah, it is called `put` but implements `upsert` functionality which means updates (if already exists) or inserts if not already exists.

File: docs/06_how-to-guides/30_key-value-api/40_how-to-delete-from-table.md (hunk 2 above)
Comment (iamveritas) [reply docs/06_how-to-guides/30_key-value-api/40_how-to-delete-from-table.md:78]: to delete the `erase` of kv::table is used. which receives as input param an object stored in the kv::table not a key.

File: docs/06_how-to-guides/30_key-value-api/40_how-to-delete-from-table.md (hunk 1 above)
Comment (iamveritas) [reply docs/06_how-to-guides/30_key-value-api/40_how-to-delete-from-table.md:76]: thanks!

File: docs/06_how-to-guides/30_key-value-api/40_how-to-delete-from-table.md (hunk 2 above)
Comment (iamveritas) [reply docs/06_how-to-guides/30_key-value-api/40_how-to-delete-from-table.md:78]: scratch the above, I was looking at the old syntax.
thank you

**Review by kimjh2005 — APPROVED** [review]

**Pull request merged**